vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -693,6 +693,13 @@ private void reset() {
                 runBeforeClientResponse(sizeRequest);
             }
         }
+
+        Set<Object> currentItemIds = getCheckboxItems()
+                .map(CheckBoxItem::getItem).map(this::getItemId)
+                .collect(Collectors.toSet());
+        getValue().stream()
+                .filter(item -> !currentItemIds.contains(getItemId(item)))
+                .forEach(this::deselect);
```

Comment:
All the tests pass for me locally even if I remove this code

Reply:
Removed in the latest commit.